src/java.base/share/classes/java/time/format/DateTimeFormatterBuilder.java

```diff
@@ -4197,6 +4197,8 @@ protected PrefixTree getTree(DateTimeParseContext context) {
                 (tree = entry.getValue().get()) == null)) {
                 tree = PrefixTree.newTree(context);
                 zoneStrings = TimeZoneNameUtility.getZoneStrings(locale);
+                Set<String> nonRegionIds = new HashSet<>(64);
```

Comment:
Does the same argument about load factor applies here? So it should be not `64`, but something more? Load factor `0.75` implies it should be at least `86`. (The size would be rounded to the next power of two internally, so `128` might as well do.)

Reply:
It's not clear why 64 was chosen here. Reading https://bugs.openjdk.org/browse/JDK-8235238 I don't see anything which particularly justifies 64. On my system:

```
Set<String> regionIds = ZoneRulesProvider.getAvailableZoneIds();
Set<String> nonRegionIds = new HashSet<>();
for (String[] z : TimeZoneNameUtility.getZoneStrings(Locale.getDefault());) {
    if (!regionIds.contains(z[0])) nonRegionIds.add(z[0]);
}
System.out.println(nonRegionIds.size());
```

gives 28, so it seems 64 already accounts for load factor nicely at least on my system.

Reply:
Right. Actually, scratch my comment, since the current version matches what mainline does.

Reply:
While I'm happy you're happy, that's not quite right - mainline uses `HashSet.newHashSet(64)` which is equivalent to `new HashSet(128)`, but I don't think that was a justified change in mainline.